guides/auth-torii-with-github.md

```diff
@@ -0,0 +1,581 @@
+`401 (Unauthorized)` for an API call to `https://api.github.com/user`. This particular API uses the identity associated
+with the authorized user to return the profile for that user, which it can't do if you're not authorized. Now that we
+have evidence that we're not truly authorized, let's fix it.
+
+We need to have a token exchange service, a service that takes an authorization code and returns an access token. You
+can implement this in any language and stack you want for writing APIs. I chose to use AWS API Gateway and Lambda. I
+won't go through the whole process of setting up the API here. AWS has good documentation for most of the process and
+leads you through it pretty well. I documented a gotcha to watch for
+[here](https://blog.vance.com/aws-api-gateway-lambdas-and-cors-cfb98517a21a). My lambda looks like
+
+```js
+'use strict';
+
+console.log('Loading function');
+
+const https = require('https');
+
+exports.handler = (event, context, callback) => {
+    console.log('Received event:', JSON.stringify(event, null, 2));
+
+    const done = (err, res) => callback(null, {
+        statusCode: err ? '400' : '200',
+        body: err ? err.message : JSON.stringify(res),
+        headers: {
+            'Content-Type': 'application/json',
+            'Access-Control-Allow-Origin': '*'
+        },
+    });
+
+    if (event.httpMethod !== 'POST') {
+        done(new Error(`Unsupported method "${event.httpMethod}"`));
+    }
+    
+    const eventBody = JSON.parse(event.body);
+    
+    const payload = {
+        'client_id': process.env.CLIENT_ID,
+        'client_secret': process.env.CLIENT_SECRET,
+        'code': eventBody.authorizationCode
+    };
+    if (eventBody.state) {
+        payload.state = eventBody.state;
+    }
+    
+    const data = JSON.stringify(payload);
+    
+    const options = {
+        hostname: 'github.com',
+        port: 443,
+        path: '/login/oauth/access_token',
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': Buffer.byteLength(data),
+            'Accept': 'application/json',
+            'User-Agent': process.env.USER_AGENT
+        }
+    };
+
+    const req = https.request(options, (res) => {
+        let body = '';
+        console.log('Status:', res.statusCode);
+        console.log('Headers:', JSON.stringify(res.headers));
+        res.setEncoding('utf8');
+        res.on('data', (chunk) => body += chunk);
+        res.on('end', () => {
+            console.log('Successfully processed HTTPS response');
+            // If we know it's JSON, parse it
+            if (res.headers['content-type'] === 'application/json') {
+                body = JSON.parse(body);
+            }
+            console.log(`Successful response: ${body}`);
+            done(null, body);
+        });
+    });
+    req.on('error', callback);
+    req.write(data);
+    req.end();
+};
+```
```

Comment:
It might make sense to show how to do this with Ember CLI's mock server for the development environment as well.